hackage-security/src/Hackage/Security/Util/FileLock.hsc

```diff
@@ -1,6 +1,8 @@
 {-# LANGUAGE CPP #-}
 {-# LANGUAGE InterruptibleFFI #-}
 {-# LANGUAGE DeriveDataTypeable #-}
+{-# LANGUAGE LambdaCase #-}
```

Comment:
LambdaCase and MultiWayIf are only GHC-7.6+,  we test for GHC-7.4

I'd rather not use these, the patch would be less controversial then.